cachito/web/metrics.py

```diff
@@ -0,0 +1,25 @@
+import os
+
+from prometheus_client import multiprocess
+from prometheus_client.core import CollectorRegistry
+from prometheus_flask_exporter.multiprocess import GunicornInternalPrometheusMetrics
+
+
+def init_metrics(app):
+    """
+    Initialize the Prometheus Flask Exporter.
+
+    :return: a Prometheus Flash Metrics object
+    :rtype: PrometheusMetrics
+    """
+    registry = CollectorRegistry()
+
+    multiproc_temp_dir = app.config["PROMETHEUS_METRICS_TEMP_DIR"]
+
+    if not os.path.isdir(multiproc_temp_dir):
+        os.makedirs(multiproc_temp_dir)
+    multiprocess.MultiProcessCollector(registry, path=multiproc_temp_dir)
+    metrics = GunicornInternalPrometheusMetrics.for_app_factory(
+        group_by="endpoint", defaults_prefix="cachito_flask"
+    )
```

Comment:
Why does it have to be the `GunicornInternal` one?

Reply:
Short, short answer - because it works.   Longer version, because using MultiprocessPrometheusMetrics we don't get the /metrics endpoint exposed, even with a self.register_endpoint("/metrics") added in. I've gone over the code for the package, I don't understand why.   Looking at the code for the GunicordInternal class, I don't see any harm in leaving this.   If you want to go over this together and try to figure out why I can't get it to work, I'm all for it.